test/undo_index.cpp

```diff
@@ -9,8 +9,8 @@
 
 // gcc-11 adds new dynamic memory warnings that return false positives for many of the stack allocated instantiations of
 // chainbase::undo_index<>. For example, see test_insert_modify. Evaluation of the actual behavior using ASAN with gcc-11 and
-// clang-11 indicates these are false positives. The warning is disabled for gcc-11 only.
-#if defined(__GNUC__) && (__GNUC__ == 11) && !defined(__clang__)
+// clang-11 indicates these are false positives. The warning is disabled for gcc-11 and above.
+#if defined(__GNUC__) && (__GNUC__ >= 11) && !defined(__clang__)
```

Comment:
Should we limit to 11 & 12 with the hope it will not be needed for gcc-13?

Reply:
We could. When I talked with Areg yesterday we did agree to turn it off for 11+ though.
I also talked with him about changing to c++20 for appbase and chainbase, but he would like me to wait on that (just fix the issues).

Reply:
It does seem like a flaky warning, and since we aren't changing chainbase much, I doubt that suppressing it prevent up from catching many errors.

Reply:
> Should we limit to 11 & 12 with the hope it will not be needed for gcc-13?

I think it would be better to take a wait and see approach with future gcc releases. If we disable it now, we'll forget this code is even here... I'd guess it's less costly to reevaluate this code block for ever gcc release than it is to devug the problem it's warning against - of course, we hope that's never an issue.

Reply:
Except we waste time for each new gcc release to investigate the warning. Believe it or not I probably spent 1/2 hour looking at chainbase code before giving up (couldn't find anything wrong), and then Areg found the comment at the top of the file.

Reply:
I do get your point, @ScottBailey , but we barely change the cainbase code at all, so it is unlikely that having this warning disabled is a big loss.